src/Nethermind/Nethermind.Network.Stats/Model/InitiateDisconnectReason.cs

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Stats.Model;
+
+/// <summary>
+/// Nethermind level disconnect reason. Don't forget to add the corresponding Eth level disconnect reason in `InitiateDisconnectReasonExtension`.
+/// </summary>
+public enum InitiateDisconnectReason : byte
+{
+    IncomingConnectionRejectedTooManyPeer,
```

Comment:
just `TooManyPeers`?

Reply:
Its too long, yes. But too many peers might disconnect at multiple place. Need to differentiate somehow.